lib/ts_utils/metadata.py

```diff
@@ -269,10 +269,6 @@ def read_metadata(distribution: str) -> StubMetadata:
         assert type(requires_python_str) is str
         requires_python = Specifier(requires_python_str)
         assert requires_python != oldest_supported_python_specifier, f'requires_python="{requires_python}" is redundant'
-        # Check minimum Python version is not less than the oldest version of Python supported by typeshed
-        assert oldest_supported_python_specifier.contains(
-            requires_python.version
-        ), f"'requires_python' contains versions lower than typeshed's oldest supported Python ({oldest_supported_python})"
```

Comment:
revert this removal too? (And maybe add a link to the discussion in this PR so it's clear why the assertion is useful for correctness?)